lib/inferno/entities/test_group.rb

```diff
@@ -64,6 +64,24 @@ def test_metadata
             repo: Inferno::Repositories::Tests.new
           }
         end
+        
+        def used_tags
+          @@used_tags ||= []
+        end
+
+        def tag(new_tag = nil)
+          @tag = new_tag.to_s unless new_tag.nil?
+          @tag ||= begin
+            tag = title.scan(/\b[A-Z]/)&.join
+            if used_tags.include? tag[0..2]
+              tag = ('A'..'Z').to_a.sample(3).join while used_tags.include? tag
```

Comment:
These ids are unstable and change each time you start Inferno. Rather than trying to generate tags based on titles, I'd recommend just using numeric tags that indicate a test's location in the suite. Then you don't have to worry about tag collisions, and the tag tells you where the test is within the suite. For example, the first test in the first group would be `1.1`, nested groups would lead to tags like `2.3.2`.

Reply:
Sounds good, that'll make more straightforward.